src/leader.cpp

```diff
@@ -21,7 +21,9 @@
 #include <unistd.h>
 #include <queue>
 #include <dirent.h>
+#if !defined(__FreeBSD__)
```

Comment:
I'd prefer `#ifdef __linux__` since this is, IIUC, a Linux-only feature rather than a everything-but-FreeBSD feature